### ISSUE TITLE
 ClusterDisruptionIT#testRestartNodeWhileIndexing test failure

### ISSUE THREAD

**Issue by tbrooks8** [elastic/elasticsearch#57850]
**Build scan**:

https://gradle-enterprise.elastic.co/s/bo7dsli36yuz6

**Repro line**:

./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.search.aggregations.pipeline.DateDerivativeIT.testSingleValuedFieldNormalised_timeZone_CET_DstStart" -Dtests.seed=742714E86BECF3AD -Dtests.security.manager=true -Dtests.locale=es-CR -Dtests.timezone=Asia/Vladivostok -Druntime.java=8

**Reproduces locally?**:

Does not reproduce

**Applicable branches**:

Has only failed on 7.x as far as I have seen.
**Failure excerpt**:

```
14:39:50 org.elasticsearch.discovery.ClusterDisruptionIT > testRestartNodeWhileIndexing FAILED
14:39:50     java.lang.AssertionError: timed out waiting for green state
14:39:50         at org.junit.Assert.fail(Assert.java:88)
14:39:50         at org.elasticsearch.test.ESIntegTestCase.ensureColor(ESIntegTestCase.java:937)
14:39:50         at org.elasticsearch.test.ESIntegTestCase.ensureGreen(ESIntegTestCase.java:890)
14:39:50         at org.elasticsearch.test.ESIntegTestCase.ensureGreen(ESIntegTestCase.java:879)
14:39:50         at org.elasticsearch.discovery.ClusterDisruptionIT.testRestartNodeWhileIndexing(ClusterDisruptionIT.java:491)
14:39:50 
14:39:50     java.lang.AssertionError: [restart_while_indexing][0], node[wF3ZginrRpSyDWh0CGBcOQ], [R], recovery_source[peer recovery], s[INITIALIZING], a[id=JkM093BGSXquOhGdFnxp-w], unassigned_info[[reason=INDEX_CREATED], at[2020-06-08T20:38:44.435Z], delayed=false, allocation_status[no_attempt]] seq_no_stats mismatch
14:39:50     Expected: <SeqNoStats{maxSeqNo=4999, localCheckpoint=4999, globalCheckpoint=4999}>
14:39:50          but: was <SeqNoStats{maxSeqNo=4999, localCheckpoint=-1, globalCheckpoint=-1}>
14:39:50         at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
14:39:50         at org.junit.Assert.assertThat(Assert.java:956)
```

**Comment by ywelsch** [comment]
The test failure here does not seem to coincide with the reproduction line (`ClusterDisruptionIT` failed, but repo line shows `DateDerivativeIT`

**Comment by ywelsch** [comment]
This looks like a peer recovery getting stuck, see

```
[2020-06-08T09:39:16,525][INFO ][o.e.d.ClusterDisruptionIT] [testRestartNodeWhileIndexing] ensureGreen timed out, cluster state:
cluster uuid: gtbU-U1TSXaUrqv2cgkxaw [committed: true]
version: 18
state uuid: QV4Crx-JRBer0paG2NgsHQ
from_diff: false
meta data version: 15
   coordination_metadata:
      term: 1
      last_committed_config: VotingConfiguration{Dx8zfi50Rkq2g1b0l2EwvQ,6zke5kOsT8KkycZ3iON0PA,wF3ZginrRpSyDWh0CGBcOQ}
      last_accepted_config: VotingConfiguration{Dx8zfi50Rkq2g1b0l2EwvQ,6zke5kOsT8KkycZ3iON0PA,wF3ZginrRpSyDWh0CGBcOQ}
      voting tombstones: []
   [restart_while_indexing/1BOAj-eLR6q8GhsiRuB5MQ]: v[14], mv[11], sv[1], av[1]
      0: p_term [1], isa_ids [mSnZ-GP9R-OUtKBIyhigog, gX3M2J6gT1-ze0iAX-zdsg]
metadata customs:
   index-graveyard: IndexGraveyard[[]]
nodes: 
   {node_t0}{6zke5kOsT8KkycZ3iON0PA}{VXUhtlsRR9SuV4UwJ790HA}{127.0.0.1}{127.0.0.1:35889}{dimr}
   {node_t2}{wF3ZginrRpSyDWh0CGBcOQ}{iQ5YOPdxRmWdZqrBSr9myQ}{127.0.0.1}{127.0.0.1:35167}{dimr}, master
   {node_t1}{Dx8zfi50Rkq2g1b0l2EwvQ}{u2uwvzTcSQOD-LMcRbWqFw}{127.0.0.1}{127.0.0.1:40697}{dimr}
routing_table (version 5):
-- index [[restart_while_indexing/1BOAj-eLR6q8GhsiRuB5MQ]]
----shard_id [restart_while_indexing][0]
--------[restart_while_indexing][0], node[Dx8zfi50Rkq2g1b0l2EwvQ], [P], s[STARTED], a[id=gX3M2J6gT1-ze0iAX-zdsg]
--------[restart_while_indexing][0], node[wF3ZginrRpSyDWh0CGBcOQ], [R], recovery_source[peer recovery], s[INITIALIZING], a[id=JkM093BGSXquOhGdFnxp-w], unassigned_info[[reason=INDEX_CREATED], at[2020-06-08T20:38:44.435Z], delayed=false, allocation_status[no_attempt]]
--------[restart_while_indexing][0], node[6zke5kOsT8KkycZ3iON0PA], [R], s[STARTED], a[id=mSnZ-GP9R-OUtKBIyhigog]

routing_nodes:
-----node_id[Dx8zfi50Rkq2g1b0l2EwvQ][V]
--------[restart_while_indexing][0], node[Dx8zfi50Rkq2g1b0l2EwvQ], [P], s[STARTED], a[id=gX3M2J6gT1-ze0iAX-zdsg]
-----node_id[wF3ZginrRpSyDWh0CGBcOQ][V]
--------[restart_while_indexing][0], node[wF3ZginrRpSyDWh0CGBcOQ], [R], recovery_source[peer recovery], s[INITIALIZING], a[id=JkM093BGSXquOhGdFnxp-w], unassigned_info[[reason=INDEX_CREATED], at[2020-06-08T20:38:44.435Z], delayed=false, allocation_status[no_attempt]]
-----node_id[6zke5kOsT8KkycZ3iON0PA][V]
--------[restart_while_indexing][0], node[6zke5kOsT8KkycZ3iON0PA], [R], s[STARTED], a[id=mSnZ-GP9R-OUtKBIyhigog]
---- unassigned
```

The second failure (`seq_no_stats mismatch`) is just a follow-up failure during clean-up phase of the test (while it already failed). 

This test failure looks like it might be addressed by https://github.com/elastic/elasticsearch/pull/57859?

**Comment by tbrooks8** [comment]
I think this is fixed by #57859.
